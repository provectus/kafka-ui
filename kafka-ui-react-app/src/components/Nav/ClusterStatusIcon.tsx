import { ServerStatus } from 'generated-sources';
import React, { CSSProperties } from 'react';

interface Props {
  status: ServerStatus;
}

const ClusterStatusIcon: React.FC<Props> = ({ status }) => {
  const style: CSSProperties = {
    width: '10px',
    height: '10px',
    borderRadius: '5px',
    marginLeft: '7px',
    padding: 0,
  };

  return (
    <span
      className={`tag ${
        status === ServerStatus.ONLINE ? 'is-success' : 'is-danger'
      }`}
      title={status}
      style={style}
    />
  );
};

export default ClusterStatusIcon;
