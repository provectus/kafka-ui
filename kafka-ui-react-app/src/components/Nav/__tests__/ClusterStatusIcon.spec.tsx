import React from 'react';
import { mount } from 'enzyme';
import { ServerStatus } from 'generated-sources';
import ClusterStatusIcon from 'components/Nav/ClusterStatusIcon';

describe('ClusterStatusIcon', () => {
  it('matches snapshot', () => {
    const wrapper = mount(<ClusterStatusIcon status={ServerStatus.ONLINE} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders online icon', () => {
    const wrapper = mount(<ClusterStatusIcon status={ServerStatus.ONLINE} />);
    expect(wrapper.exists('.is-success')).toBeTruthy();
    expect(wrapper.exists('.is-danger')).toBeFalsy();
  });
  it('renders offline icon', () => {
    const wrapper = mount(<ClusterStatusIcon status={ServerStatus.OFFLINE} />);
    expect(wrapper.exists('.is-danger')).toBeTruthy();
    expect(wrapper.exists('.is-success')).toBeFalsy();
  });
});
