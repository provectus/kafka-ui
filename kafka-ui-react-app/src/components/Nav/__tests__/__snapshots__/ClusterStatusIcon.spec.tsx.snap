// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClusterStatusIcon matches snapshot 1`] = `
<ClusterStatusIcon
  status="online"
>
  <span
    className="tag is-primary"
    style={
      Object {
        "borderRadius": "5px",
        "height": "10px",
        "marginLeft": "7px",
        "padding": 0,
        "width": "10px",
      }
    }
    title="online"
  />
</ClusterStatusIcon>
`;
